### PR TITLE
Do things with Config Newtons

### LINKED ISSUE #16: Make this project more generic
The name, provided CLA, and favicon are Salesforce specific.

### PR DESCRIPTION
**Pull request by DarthHater**
This will suck some of the hard coded values into a conf file, currently.

More to come, just putting it out there so that I can get early feedback.
 
Potentially fixes #16

### CODE REVIEW

File: app/controllers/Application.scala [hunk 1]
```diff
@@ -64,9 +64,11 @@ class Application @Inject()
 
   val gitHubOauthScopesForClaSigning = Seq("user:email")
   val gitHubOauthScopesForAudit = Seq("read:org")
+  val organizationName = configuration.get[String]("app.organization.name")
+  val organizationUrl = configuration.get[String]("app.organization.url")
```
Comment (jamesward) [inline app/controllers/Application.scala]: I think we could make both of these optional.  Then in the views leave off the name if not set.

File: app/controllers/Application.scala [hunk 2]
```diff
@@ -64,9 +64,11 @@ class Application @Inject()
...
 
   def index = Action {
-    Redirect("http://salesforce.github.io/")
+    Redirect(organizationUrl)
```
Comment (jamesward) [inline app/controllers/Application.scala]: If the optional value isn't set we could either render a generic page about the CLA or redirect to the signing page.

File: conf/application.conf [hunk 3]
```diff
@@ -23,6 +23,9 @@ github.integration.slug = ${GITHUB_INTEGRATION_SLUG}
 github.integration.private-key = ${GITHUB_INTEGRATION_PRIVATE_KEY}
 github.integration.secret-token = ${?GITHUB_INTEGRATION_SECRET_TOKEN}
 
+app.organization.name = "Salesforce"
```
Comment (jamesward) [inline conf/application.conf]: Let's make the value of this something that can pull from an env var: `${?ORG_NAME}`

File: conf/application.conf [hunk 4]
```diff
@@ -23,6 +23,9 @@ github.integration.slug = ${GITHUB_INTEGRATION_SLUG}
...
+app.organization.url = "http://salesforce.github.io/"
```
Comment (jamesward) [inline conf/application.conf]: Same: `${?ORG_URL}`  Or we ditch this and do the generic CLA page instead.

File: app/controllers/Application.scala (hunk 2 above)
Comment (DarthHater) [reply app/controllers/Application.scala]: Did some of this in: dfc4125

File: app/controllers/Application.scala (hunk 1 above)
Comment (DarthHater) [reply app/controllers/Application.scala]: Pretty sure got most of this in: dfc4125

File: conf/application.conf (hunk 3 above)
Comment (DarthHater) [reply conf/application.conf]: Did this in: dfc4125

File: conf/application.conf (hunk 4 above)
Comment (DarthHater) [reply conf/application.conf]: In: dfc4125

File: app/controllers/Application.scala (hunk 1 above)
Comment (DarthHater) [reply app/controllers/Application.scala]: Haven't adjusted the views yet, let me know if you are cool with what I did otherwise.

Comment (jamesward) [reply app/controllers/Application.scala]: We want these to be `Option[String]` so we avoid nulls.  I think there is a `getOpt` to do that.

File: app/controllers/Application.scala [hunk 5]
```diff
@@ -64,9 +64,17 @@ class Application @Inject()
...
+
+  def isEmpty(x: String) = x == null || x.isEmpty
 
   def index = Action {
-    Redirect("http://salesforce.github.io/")
+    if(organizationUrl.isEmpty()) {
+      Redirect(routes.Application.signCla())
+    } else {
+      Redirect(organizationUrl)
+    }
```
Comment (jamesward) [inline app/controllers/Application.scala]: The FP way to do this when the value is an `Option[String]` is something like:
`maybeOrganizationUrl.fold(Redirect(routes.Application.signCla()))(Redirect(_))`

Comment (DarthHater) [reply app/controllers/Application.scala]: Haha trickery! I'll give it a whirl.

Comment (DarthHater) [reply app/controllers/Application.scala]: In: 618b4bc (I think)

File: app/controllers/Application.scala (hunk 1 above)
Comment (DarthHater) [reply app/controllers/Application.scala]: In: 6988036

**Comment by jamesward** [comment]
Thanks!  This is looking good.  One last thing... When you use the `maybeBlah` in a template it will `.toString` it which on an option will look like `Some("foo")` or `None` so we actually have to transform the option to a string.  I think in most cases this means doing something like: `maybeFoo.getOrElse("")` which unfortunately get's a bit tedious but that is the nature of optional values.  :)

**Pull request merged**